### PR TITLE
[CI] Add quantization

### PR DESCRIPTION
**Pull request by sayakpaul**
# What does this PR do?

As discussed.

### CODE REVIEW

File: tests/quantization/bnb/test_mixed_int8.py [hunk 1]
```diff
@@ -369,7 +369,7 @@ def test_quality(self):
             output_type="np",
         ).images
         out_slice = output[0, -3:, -3:, -1].flatten()
-        expected_slice = np.array([0.0149, 0.0322, 0.0073, 0.0134, 0.0332, 0.011, 0.002, 0.0232, 0.0193])
+        expected_slice = np.array([0.0376, 0.0359, 0.0015, 0.0449, 0.0479, 0.0098, 0.0083, 0.0295, 0.0295])
```
Comment (sayakpaul) [inline tests/quantization/bnb/test_mixed_int8.py:372]: Because of the machine change.

File: .github/workflows/nightly_tests.yml [hunk 2]
```diff
@@ -347,6 +347,64 @@ jobs:
...
+      max-parallel: 2
+      matrix: 
+        config:
+          - backend: "bitsandbytes"
+            test_location: "bnb"
+    runs-on:
+      group: aws-g6e-xlarge-plus
+    container:
+      image: diffusers/diffusers-pytorch-cuda
+      options: --shm-size "20gb" --ipc host --gpus 0
```
Comment (sayakpaul) [inline .github/workflows/nightly_tests.yml:363]: Using a large shared memory here.

File: .github/workflows/nightly_tests.yml [hunk 3]
```diff
@@ -347,6 +347,64 @@ jobs:
...
 
+  run_nightly_quantization_tests:
+    name: Torch quantization nightly tests
+    strategy:
+      fail-fast: false
+      max-parallel: 2
+      matrix: 
+        config:
+          - backend: "bitsandbytes"
+            test_location: "bnb"
```
Comment (sayakpaul) [inline .github/workflows/nightly_tests.yml:358]: Decided to go with a matrix strategy to accommodate for future backends.

**Comment by sayakpaul** [comment]
@DN6 a gentle ping

**Review by DN6 — APPROVED** [review]

**Pull request merged**